addons/misra.py

```diff
@@ -1579,6 +1580,47 @@ def misra_17_1(self, data):
             elif token.str == 'va_list':
                 self.reportError(token, 17, 1)
 
+    def misra_17_2(self, data):
+        def find_indirect_calls(func_id, calls_map, found_calls, visited):
+            if len(calls_map) == 0:
+                return found_calls
+            for tok in calls_map.get(func_id, []):
+                call_id = tok.function.Id
+                if call_id in visited:
+                    found_calls.add(tok)
+                calls_map.get(func_id).remove(tok)
+                visited.add(func_id)
+                found_calls.update(find_indirect_calls(call_id, calls_map, found_calls, visited))
+            return found_calls
+
+        # function id : list of tokens with its calls
+        function_calls = {f.Id: [] for f in data.functions}
+        for token in data.tokenlist:
+            if token.scope.type != 'Function':
+                continue
+
+            scope = token.scope
```

Comment:
this will be executed on each token in every function. It should be executed once for each function right? I suggest that you loop through the scopes in the for loop above instead of the tokens.